### DISCUSSION TITLE
Sound Wave Preview

### DISCUSSION THREAD

**Discussion by 4l3m4oO** [friction2d/friction#45] [Ideas]
![image](https://github.com/friction2d/friction/assets/357488/2b037bf6-33d9-49a3-b4ca-3419c8d379b9)

I think it will be nice to implement previews of the sounds with sound waves.

Example:
![image](https://github.com/friction2d/friction/assets/357488/c361ae81-2a11-433b-9398-7f587b8d1bfd)

**Comment by ArrayBolt3** [comment]
This would be awesome for making music lyric videos with this (which is my intended use case). I think I'll be able to live without it for now, but I'd definitely second this!

**Comment by rodlie** [comment]
Been requested in the past. The timeline (audio track) was not designed for this type of feature, so a major refactor/rewrite will need to happen to support this, will probably not happen until v2 (may happen earlier, depends on priority). Sorry about that.

**Reply by ArrayBolt3** [reply]
No problem! I'm able to use Audacity to let me convert audio positions to precise timestamps, which I can then multiply by 30 to get frame number in a 30fps video. That's ended up even more precise than a sound wave preview would have been.

**Reply by pgilfernandez** [reply]
> No problem! I'm able to use Audacity to let me convert audio positions to precise timestamps, which I can then multiply by 30 to get frame number in a 30fps video. That's ended up even more precise than a sound wave preview would have been.

Nice technique, is there any tutorial you know on the subject?

**Reply by ArrayBolt3** [reply]
Sadly I don't, I have quite a bit of experience with Audacity so I mostly just figured this out.
